src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+                        x=x,
+                        y=self.results[0].rates[i].rate,
+                        name=self.results[0].rates[i].name,
+                    )
+                )
+            fig.update_layout(title_text='Rates', showlegend=True)
+            fig.update_xaxes(title_text=x_text)
+            fig.update_yaxes(title_text='rates (g product/g cat/h)')
+            self.figures.append(
+                PlotlyFigure(label='Rates', figure=fig.to_plotly_json())
+            )
+        if not self.results[0].reactants_conversions:
+            return
+        if self.results[0].reactants_conversions[0].conversion and (
+            self.results[0].products[0].selectivity
+        ):
+            for i, c in enumerate(self.results[0].reactants_conversions):
+                name = self.results[0].reactants_conversions[i].name
+                fig = go.Figure()
+                for j, p in enumerate(self.results[0].products):
+                    fig.add_trace(
+                        go.Scatter(
+                            x=self.results[0].reactants_conversions[i].conversion,
+                            y=self.results[0].products[j].selectivity,
+                            name=self.results[0].products[j].name,
+                            mode='markers',
+                        )
+                    )
+                fig.update_layout(title_text='S-X plot ' + str(i), showlegend=True)
+                fig.update_xaxes(title_text=name + ' Conversion (%)')
+                fig.update_yaxes(title_text='Selectivity (%)')
+                self.figures.append(
+                    PlotlyFigure(
+                        label='S-X plot ' + name + ' Conversion',
+                        figure=fig.to_plotly_json(),
+                    )
+                )
+
+    def normalize_reaction_conditions(
+        self, archive: 'EntryArchive', logger: 'BoundLogger'
+    ) -> None:
+        if self.reaction_conditions is not None:
+            reagents = []
+            for reagent in self.reaction_conditions.reagents:
+                if reagent.pure_component is None or reagent.pure_component == []:
+                    reagent.normalize(archive, logger)
+                reagents.append(reagent)
+            self.reaction_conditions.reagents = reagents
+
+            if self.reactor_filling is not None:
+                if (
+                    self.reaction_conditions.set_total_flow_rate is not None
+                    and self.reactor_filling.catalyst_mass is not None
+                    and self.reaction_conditions.weight_hourly_space_velocity is None
+                ):
+                    self.reaction_conditions.weight_hourly_space_velocity = (
+                        self.reaction_conditions.set_total_flow_rate
+                        / self.reactor_filling.catalyst_mass
+                    )
+
+    def return_conversion_results(
+        self, archive: 'EntryArchive', logger: 'BoundLogger'
+    ) -> None:
```

Comment:
You are returning a list and not `None`.
```suggestion
    ) -> list:
```